src/nnbench/types/types.py

```diff
@@ -164,11 +175,41 @@ class Benchmark:
     tearDown: Callable[..., None] = field(repr=False, default=NoOp)
     tags: tuple[str, ...] = field(repr=False, default=())
     interface: Interface = field(init=False, repr=False)
+    state: State | None = field(default=None)
 
     def __post_init__(self):
         if not self.name:
             super().__setattr__("name", self.fn.__name__)
         super().__setattr__("interface", Interface.from_callable(self.fn, self.params))
+        if self.state is None:
+            super().__setattr__(
+                "state",
+                State(
+                    name=self.name or "",
+                    function=self.fn,
+                    family=self.fn.__name__,
+                    family_size=1,
+                    family_index=0,
+                ),
+            )
```

Comment:
Is it sufficiently clear that this is the default if the Benchmark state is not set? that is the case for standalone benchmarks (which do not come out of `parametrize` or `product`). Therefore `family_size=1` and `family_index=0`.